index.js

```diff
@@ -1,3 +1,10 @@
 'use strict';
 module.exports = typeof setImmediate === 'function' ? setImmediate :
-	function setImmediate(fn) { setTimeout(fn, 0); };
+  function setImmediate() {
+    var args = Array.prototype.slice.apply(arguments);
+    var fn = args.shift();
+    console.log('fn', fn, 'args', args);
+    setTimeout(function() {
```

Comment:
but it still changes the arguments: `setTimeout(func, delay, [param1, param2, ...]);` so that we would have to insert the delay as the second argument right?